tests/modules/services/barrier/basic-configuration.nix

```
@@ -0,0 +1,17 @@
+{ config, pkgs, ... }:
+
+{
+  config = {
+    services.barrier.client = {
+      enable = true;
+      server = "testServer";
+    };
+
+    nmt.script = ''
+      clientServiceFile=home-files/.config/systemd/user/barrierc.service
+
```

I think everything looks nice. Just need to add
```suggestion
    nixpkgs.overlays =
      [ (self: super: { barrier = pkgs.writeScriptBin "dummy-barrier" ""; }) ];

```
so that the tests don't download the whole package (and all its dependencies).